### DISCUSSION TITLE
Initial thoughts

### DISCUSSION THREAD

**Discussion by SparkyNZ** [gusmanb/logicanalyzer#149] [Ideas]
I'm quite impressed with this overall. I have used Salae Logic quite a bit in the past but what's always put me off is of course the 16 channel limitation.

What I feel is missing is the support of a clock signal. Right now I have a slow-stepping 6502 system and I only get to see 1 transition when I use the pattern trigger. Because the fixed sampling frequency is so high, I see nothing else. Will you be adding support for a clock signal in the future?

The only other thing for me is the need for a protocol analyser that would show an 16-bit address and 8-bit data bus as hex. I get the feeling I could write a custom one (I am more a software dev than an electronics guru anyway) but I wouldn't go down that path without the support of a clock signal.

**Comment by gusmanb** [comment]
Sorry, what do you mean with "fixed sampling frequency"? The frequency is configurable from 3Khz~ to 100Mhz (and up to 400Mhz in V6).

In any case, you can emulate the clock behavior with the burst mode, configure an edge trigger, set the post-capture to 2-4 samples (base it on your real clock), enable the burst mode and that's it, you get the same behavior as having a clock signal.

Regarding protocols, in V6.0 you will have every analyzer from Sigrok and you could even write your own ones in Python.

Cheers.

**Reply by SparkyNZ** [reply]
Sorry I haven't had a chance to have another look (busy life :-) ). By "fixed sampling frequency" I was referring to the 100,000,000Hz frequency shown in the bottom right of the Logic Analyzer application. I didn't realise it was possible to change this. I couldn't find any settings/prefs in the menus to change it.

I'll try what you were saying with the burst mode when I get a chance. BTW, my hardware is v4.0. I would be interested in assembling a second v4.0 board and linking them together. I hope I'll still be able to use the v4.0 boards with the v6.x Logic Analyzer application. I think my current version is 5.1

It would be great to have "rollover feedback" in the Logic Analyzer too. A lot of the time I can't make out which signals are low and which signals are high. Would be nice to show a 0 or 1 when hovering the mouse pointer over a signal perhaps.

Thanks for the quick reply - sorry I only get the chance to fiddle with my projects on weekends if I'm lucky.

**Reply by gusmanb** [reply]
There is no menu for settings, when you start a capture the capture dialog contains all the settings for the capture, there is where you set the sampling frequency.

![imagen](https://github.com/user-attachments/assets/41943b1e-df54-4881-9820-4e0d59076854)


Yes, V6.0 is compatible with previous boards, the new board does not change the design too much, is an "all in one" with some patches to support the Pico 2.

The hovering is already implemented in V6.0, when you put your mouse over a signal it tells you if it is high or low and for how many time/samples.

![imagen](https://github.com/user-attachments/assets/a7d94f3b-7907-410d-b73c-4c8a8a653b7c)

Cheers.

**Reply by SparkyNZ** [reply]
Hi. I've given this another try - sorry it took so long. I'm still missing something:
https://imgur.com/7w3Q095

I have the clock connected to Channel 24. My clock LED is currently flashing every second. When I start the capture, it seems to trigger on one of the clock pulses but then it stops and doesn't capture any more (its only showing 1536 total samples in the bottom right of the screen). How can I configure it to capture, say, 20 clock cycles?

I wasn't sure if you meant configure the Burst count to 4 or Postsamples. The post samples are just Frequency samples that are counted after the Edge trigger starts I guess? I also can't seem to set the Frequency to lower than 31000.

I'll leave the analyser connected to my circuit so I can respond a bit sooner to you next time.

![image](https://github.com/user-attachments/assets/aa2f4407-7323-4d06-b8ad-b8272dff0a20)

**Reply by gusmanb** [reply]
Hi.

The frequency on that version is correct, my bad, I missed one zero, the minimum is around 30Khz, but in the V6.0 it can go down up to 3Khz~ with the non-turbo version.

Regarding the burst, are you sure that not all samples are captured? That version does not show the total ammount computed with the bursts, check if the end of the capture reaches to 4608 samples. If your clock is so slow then is possible that the trigger gets rearmed and detects immediatelly the trigger condition again and continues the capture.

Try this: download the V6.0 version from [here](https://github.com/gusmanb/logicanalyzer/discussions/148) and update the firmware to the non-turbo one. You will be able to set now the frequency to 3103 samples if I recall it right. Then configure the presamples to 10, postsamples to 3200, the burst count to 20, enable the burst mode and the burst measure. With that it will capture the full "ON" of the clock and a bit of the off, it will stop and on the next ON it will trigger again. Also, it will have a marker telling you how much time passed since it re-triggered each burst (you will have a marker atop the sample marker bar).

**Reply by SparkyNZ** [reply]
Thanks again for the quick response. I just ducked out of my job to give this a try :-) Just having a few issues setting up the capture. I couldn't set the Frequency to 3,000 - it changes it to 31,000 but I managed to get 3,100 to stay there. Only snag now is that I'm being told "Total samples cannot exceed 32768".

![image](https://github.com/user-attachments/assets/51e430f6-96c0-40e6-96f1-5571ccd71b20)

**Reply by gusmanb** [reply]
Well, you are telling the device to capture 10 + (3,200 * 20), that's 64,010 samples, so it's more than 32,768... (my bad, when you told me what you wanted to capture didn't noticed how many channels you were using, you are using the 24 channel mode so the max samples is 128k / 4, you can check the limits of your device if you push the "i" near the device name)😄

Reduce the burst count to 10 or try to use one less channel and use only the first 16, that will place the device in 16 channel mode.

Cheers.

**Reply by SparkyNZ** [reply]
Yay! That seems to work. Its a shame we couldn't lower the sample frequency even more so I could get more total samples. If I turn my clock frequency up then I get loads of data (which is what we'd expect on "real" computer system :-) ) Is it not possible to allow a really low sampling frequency perhaps in a later version? Something silly like 10Hz would do the trick.

The only other observation I have is that on my 4k monitor I'm having to use the entire screen height to see the 24 channels. It would be great to be able to alter the heights of each channel row - either by dragging (more effort) or by having a setting somewhere to set the size as "half, quarter, eighth or whatever"... 1/x would be good - and you'd have to watch out for some silly user putting in a 0 ;-)

![image](https://github.com/user-attachments/assets/cc4290bb-7da5-41ae-853b-57ebba8772a9)

**Reply by SparkyNZ** [reply]
I just reloaded the saved .lac file and had an exception too. Probably something you're already aware of. Let me know you want a copy of the .lac file or if you'd like me to create a separate issue or whatever. I know v6 is still work in progress so no worries from me - just FYI:
![image](https://github.com/user-attachments/assets/0e557892-8044-4e8c-94d4-7f12a97225e5)
